src/cloud-api-adaptor/test/provisioner/provision.go

```diff
@@ -530,17 +543,7 @@ func (p *CloudAPIAdaptor) Deploy(ctx context.Context, cfg *envconf.Config, props
 	}
 
 	for ds, timeout := range daemonSetList {
-		// Wait for the daemonset to have at least one pod running then wait for each pod
-		// be ready.
-
-		fmt.Printf("Wait for the %s DaemonSet be available\n", ds.GetName())
-		if err = wait.For(conditions.New(resources).ResourceMatch(ds, func(object k8s.Object) bool {
-			ds = object.(*appsv1.DaemonSet)
-
-			return ds.Status.CurrentNumberScheduled > 0
-		}), wait.WithTimeout(time.Minute*5)); err != nil {
-			return err
-		}
```

Comment:
Why has this been deleted? Can we have some more information about the motivation for the changes in the commit message please?

Reply:
Added a comments and leave this section as commented, we can clean it later.

Reply:
So is the argument that if we are waiting for the pods, then we don't need to wait for the ds first?
Is there a small window where the ds is not ready, so the pods haven't been created yet? Is there any harm in waiting for the ds to be ready first?

Reply:
Yes, I met problem several times when waiting the DS status indeed in libvirt provider, I'm guessing the interface is designed for deploy instead of DS. Let me try more and find more details.

Reply:
It hangs on here in my env:
```
make test-e2e
go test -v -tags=libvirt -timeout 60m -count=1 ./test/e2e
time="2024-05-08T08:51:13Z" level=info msg="Do setup"
time="2024-05-08T08:51:13Z" level=info msg="Deploying kbs"
time="2024-05-08T08:51:13Z" level=info msg="creating key.bin"
/root/src/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/test/e2e
time="2024-05-08T08:51:13Z" level=info msg="Creating kbs install overlay"
time="2024-05-08T08:51:13Z" level=info msg="Customize the overlay yaml file"
time="2024-05-08T08:51:13Z" level=info msg="Updating kbs image with \"ghcr.io/confidential-containers/staged-images/kbs\""
time="2024-05-08T08:51:13Z" level=info msg="Updating kbs image tag with \"dc01f454264fb4350e5f69eba05683a9a1882c41\""
time="2024-05-08T08:51:13Z" level=info msg="Creating kbs install overlay"
time="2024-05-08T08:51:13Z" level=info msg="Install Kbs"
Wait for the kbs deployment be available
time="2024-05-08T08:51:18Z" level=info msg="KBS PARAMS: cc_kbc::http://192.168.122.123:30205"
time="2024-05-08T08:51:18Z" level=info msg="Enable customized opa file in KBS service."
time="2024-05-08T08:51:18Z" level=info msg=EnableKbsCustomizedPolicy
time="2024-05-08T08:51:18Z" level=info msg="Install Cloud API Adaptor"
time="2024-05-08T08:51:18Z" level=info msg="Deploy the Cloud API Adaptor"
time="2024-05-08T08:51:18Z" level=info msg="Install the controller manager"
Wait for the cc-operator-controller-manager deployment be available
time="2024-05-08T08:51:36Z" level=info msg="Customize the overlay yaml file"
time="2024-05-08T08:51:38Z" level=info msg="Install the cloud-api-adaptor"
Wait for the cc-operator-daemon-install DaemonSet be available
```
Checking from another console, like this:
```
# kubectl -n confidential-containers-system get ds
NAME                           DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                            AGE
cc-operator-daemon-uninstall   0         0         0       0            0           katacontainers.io/kata-runtime=cleanup   12m
cloud-api-adaptor-daemonset    1         1         1       1            1           node.kubernetes.io/worker=               11m
```
With the change in the PR, it works well. Looks its because the DESIRED is 0 for `cc-operator-daemon-uninstall`?
Maybe, I can roll it back and do more investigation on this issue as it's not related with this PR.

Reply:
Ah ok - I have seen timeouts on it, but when I previously checked it was just that the libvirt system was slow to get going, so just re-run it if it fails and the next time it always works? We could try increasing the timeout first to see if that makes a difference?

Reply:
oh, I think I got the problem :-), dropping a fix...

Reply:
it's because no `cc-operator-daemon-install` ds installed in my cluster.

Reply:
I'll roll it back and check why I don't have cc-operator-daemon-install installed.